src/Domain/Creators/TimeDependence/Shape.cpp

```diff
@@ -0,0 +1,151 @@
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+
+namespace domain {
+namespace creators::time_dependence {
+using SphereTransition =
+    domain::CoordinateMaps::ShapeMapTransitionFunctions::SphereTransition;
+
+Shape::Shape(const double initial_time, const size_t l_max, const double mass,
+             const std::array<double, 3> spin,
+             const std::array<double, 3> center, const double inner_radius,
+             const double outer_radius, const Options::Context& context)
+    : initial_time_(initial_time),
+      l_max_(l_max),
+      mass_(mass),
+      spin_(spin),
+      center_(center),
+      inner_radius_(inner_radius),
+      outer_radius_(outer_radius),
+      transition_func_(std::make_unique<SphereTransition>(
+          SphereTransition{inner_radius, outer_radius})) {
+  if (magnitude(spin) >= 1.0) {
+    PARSE_ERROR(context,
+                "Tried to create a Shape TimeDependence, but "
+                "the magnitude of the spin ("
+                    << spin << ") is greater than one.");
+    // There is no PARSE_ERROR for the `inner_radius` < `outer_radius` condition
+    // because the SphereTransition already checks for this condition.
+  }
+}
+
+std::unique_ptr<TimeDependence<3>> Shape::get_clone() const {
+  return std::make_unique<Shape>(initial_time_, l_max_, mass_, spin_, center_,
+                                 inner_radius_, outer_radius_);
+}
+
+std::vector<
+    std::unique_ptr<domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, 3>>>
+Shape::block_maps_grid_to_inertial(
+    const size_t number_of_blocks) const {
+  ASSERT(number_of_blocks > 0,
+         "Must have at least one block on which to create a map.");
+
+  std::vector<std::unique_ptr<
+      domain::CoordinateMapBase<Frame::Grid, Frame::Inertial, 3>>>
+      result{number_of_blocks};
+  result[0] = std::make_unique<GridToInertialMap>(grid_to_inertial_map());
+  for (size_t i = 1; i < number_of_blocks; ++i) {
+    result[i] = result[0]->get_clone();
+  }
+  return result;
+}
+
+std::unordered_map<std::string,
+                   std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+Shape::functions_of_time(
+    const std::unordered_map<std::string, double>& initial_expiration_times)
+    const {
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      result{};
+
+  // Functions of time don't expire by default
+  double expiration_time = std::numeric_limits<double>::infinity();
+
+  // If we have control systems, overwrite the expiration time with the one
+  // supplied by the control system
+  if (initial_expiration_times.count(function_of_time_name_) == 1) {
+    expiration_time = initial_expiration_times.at(function_of_time_name_);
+  }
+
+  // This is Kerr r_+, the radius of the outer horizon.
+  const double horizon_boyer_lindquist =
+      mass_ * (1. + sqrt(1. - dot(spin_, spin_)));
```

Comment:
Haven't looked at this carefully yet, but I think this always conforms to a Kerr horizon. We excise a bit inside the horizon, so (at least for initial data solves, but I think also for evolutions?) we need to conform the excision to an ellipsoid of _any_ constant Boyer-Lindquist radius r_BL, not necessarily r_BL=r_+. The r_BL is given by the `inner_radius` parameter of the excision. Here's the commit on my branch that fixes that: https://github.com/nilsvu/spectre/commit/494b47479c963942ffe12a412a8e0d712a7f0403
There's also a description in [my thesis](https://publishup.uni-potsdam.de/opus4-ubp/frontdoor/deliver/index/docId/56226/file/vu_diss.pdf) (Sec. 4.1.2) that might help. (It's not particularly complicated, I just haven't seen it written down anywhere. I should really move it into the spectre docs.)

Reply:
I think we want a generic constant BL radius for more complicated domains like BBH, but in a time dependence that will most likely be used only with the shell domain, I don't know if it's that important.

Reply:
Thanks for this! Should this change be implemented here? I know the fix relies on a `kerr_radius` function that is not on develop yet, should that also be added? Or maybe in a future PR?

Reply:
If it needs something that's not on develop yet, I'd say defer the change to a future PR, but up to @nilsvu.